### PR TITLE
Onboarding for terraform-rhcs-rosa

### PR DESCRIPTION
**Pull request by enriquebelarte**

### CODE REVIEW

File: ci-operator/config/terraform-redhat/terraform-rhcs-rosa/terraform-redhat-terraform-rhcs-rosa-main.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+build_root:
+  image_stream_tag:
+    name: release
+    namespace: openshift
+    tag: golang-1.19
+images:
+- dockerfile_path: Dockerfile
+  to: clis-img
```
Comment (osherdp) [inline ci-operator/config/terraform-redhat/terraform-rhcs-rosa/terraform-redhat-terraform-rhcs-rosa-main.yaml:8]: what ``clis-img`` means?

File: ci-operator/config/terraform-redhat/terraform-rhcs-rosa/terraform-redhat-terraform-rhcs-rosa-main.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+resources:
+  '*':
+    limits:
+      memory: 4Gi
+    requests:
+      cpu: 100m
+      memory: 200Mi
+tests:
+- as: verify-format
+  commands: make init && make verify
```
Comment (osherdp) [inline ci-operator/config/terraform-redhat/terraform-rhcs-rosa/terraform-redhat-terraform-rhcs-rosa-main.yaml:18]:
```suggestion
  commands: make init verify
```
or even better, make ``verify`` depend on ``init`` target

File: ci-operator/jobs/terraform-redhat/terraform-rhcs-rosa/terraform-redhat-terraform-rhcs-rosa-main-presubmits.yaml [hunk 3]
```diff
@@ -0,0 +1,149 @@
+presubmits:
```
Comment (osherdp) [inline ci-operator/jobs/terraform-redhat/terraform-rhcs-rosa/terraform-redhat-terraform-rhcs-rosa-main-presubmits.yaml:1]: why not rehearsing the new jobs? (seeing you've already committed for "pj-rehearse ack")

**Review by osherdp — CHANGES_REQUESTED** [review]

File: ci-operator/config/terraform-redhat/terraform-rhcs-rosa/terraform-redhat-terraform-rhcs-rosa-main.yaml (hunk 2 above)
Comment (enriquebelarte) [reply ci-operator/config/terraform-redhat/terraform-rhcs-rosa/terraform-redhat-terraform-rhcs-rosa-main.yaml:18]: Do you mean changing it at the component Makefile?

File: ci-operator/config/terraform-redhat/terraform-rhcs-rosa/terraform-redhat-terraform-rhcs-rosa-main.yaml (hunk 1 above)
Comment (enriquebelarte) [reply ci-operator/config/terraform-redhat/terraform-rhcs-rosa/terraform-redhat-terraform-rhcs-rosa-main.yaml:8]: It is the name for the image containing the necessary clients for future tests.

File: ci-operator/config/terraform-redhat/terraform-rhcs-rosa/terraform-redhat-terraform-rhcs-rosa-main.yaml (hunk 2 above)
Comment (osherdp) [reply ci-operator/config/terraform-redhat/terraform-rhcs-rosa/terraform-redhat-terraform-rhcs-rosa-main.yaml:18]: yes,
```makefile
init:
    ...
verify: init
   ...
```

File: ci-operator/config/terraform-redhat/terraform-rhcs-rosa/terraform-redhat-terraform-rhcs-rosa-main.yaml (hunk 1 above)
Comment (osherdp) [reply ci-operator/config/terraform-redhat/terraform-rhcs-rosa/terraform-redhat-terraform-rhcs-rosa-main.yaml:8]: let's rename to either one of ``terraform-rhcs-rosa-clients``, ``terraform-rhcs-rosa-build``, ``clients``
naming is hard, but it's just too much to use both acronym pluralized and an abbreviation, and it interferes with reading the code

**Comment by enriquebelarte** [comment]
/pj-rehearse pull-ci-terraform-redhat-terraform-rhcs-rosa-main-verify-format

**Comment by enriquebelarte** [comment]
/pj-rehearse pull-ci-terraform-redhat-terraform-rhcs-rosa-main-verify-format

**Review by osherdp — APPROVED** [review]

**Pull request merged**